### PR TITLE
Allow a KMS Action to specify keys in the Resources of a policy

### PR DESCRIPTION
**Pull request by marktheunissen**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

With this PR, policies that specify resources for KMS actions will match on those resources, e.g:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "kms:CreateKey",
                "kms:KeyStatus",
                "kms:ListKeys"
            ],
            "Resource": [
                "arn:minio:kms:::keys-abc-*",
                "arn:minio:kms:::myuser-*"
            ]
        }
    ]
}
```

This PR also adds tests for all handlers that are involved with KMS and adds some test coverage that wasn't there before

## Motivation and Context

`kms:` actions currently disregard Resources, users have asked for restrictions to be applied to these KMS actions.

## How to test this PR?

- Needs https://github.com/minio/pkg/pull/121
- Apply a policy as above
- `mc admin kms key create ALIAS KEYNAME`
- `mc admin kms key status ALIAS KEYNAME`
- `mc admin kms key list ALIAS`

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -261,3 +261,5 @@ require (
 	google.golang.org/grpc v1.65.0 // indirect
 	google.golang.org/protobuf v1.34.2 // indirect
 )
+
+replace github.com/minio/pkg/v3 => github.com/marktheunissen/pkg v0.0.0-20240712-307f039
```
Comment (marktheunissen) [inline go.mod]: Remove before merge

**Review by klauspost — COMMENTED** [review]
👍🏼 LGTM. Will do final review when upstream is merged and this is out of Draft.

File: go.mod [hunk 2]
```diff
@@ -261,3 +261,5 @@ require (
 	google.golang.org/grpc v1.65.0 // indirect
 	google.golang.org/protobuf v1.34.2 // indirect
 )
+
+replace github.com/minio/pkg/v3 => github.com/marktheunissen/pkg/v3 v3.0.0-20240715013114-8e0d6a8ddddd
```
Comment (harshavardhana) [inline go.mod]: You can point to master now

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by marktheunissen** [comment]
~We need a new tag (v3.0.8) on pkg repo, is that a process done manually or by a bot somewhere?~

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]

**Pull request merged**